src/Dofs/sparsity_pattern.jl

```diff
@@ -1,83 +1,473 @@
+###########################
+# AbstractSparsityPattern #
+###########################
+
+"""
+    AbstractSparsityPattern
+
+Supertype for sparsity pattern implementations, e.g. [`SparsityPattern`](@ref) and
+[`BlockSparsityPattern`](@ref).
+"""
+abstract type AbstractSparsityPattern end
+
+"""
+    n_rows(sp::AbstractSparsityPattern)
```

Comment:
Is there any reason to use a new convention for the function naming here?(xref https://github.com/Ferrite-FEM/Ferrite.jl/issues/741) Maybe `getnrows`, `numrows`, or at least `nrows` to be somewhat consistent in the naming?

Analogue for the 2 functions below.